include/bm/bm_sim/named_p4object.h

```diff
@@ -46,9 +51,13 @@ class NamedP4Object {
   //! Get the compiler-provided id
   p4object_id_t get_id() const { return id; }
 
+  bool has_source_info() const { return (source_info != nullptr); }
+  const SourceInfo *get_source_info() const { return source_info; }
+
  protected:
   const std::string name;
   p4object_id_t id;
+  const SourceInfo *source_info;
```

Comment:
UPDATE: I figure out one way to make unique_ptr's work, which is now part of this PR.  You may still have comments about how I have used unique_ptr's, if I have done so incorrectly or in an odd way.